src/main/groovy/com/bugsnag/android/gradle/BugsnagMultiPartUploadTask.groovy

```diff
@@ -97,17 +94,18 @@ abstract class BugsnagMultiPartUploadTask extends BugsnagVariantOutputTask {
         httpPost.setEntity(mpEntity)
 
         HttpClient httpClient = new DefaultHttpClient()
-        HttpParams params = httpClient.getParams()
+        HttpParams params = httpClient.params
         HttpConnectionParams.setConnectionTimeout(params, TIMEOUT_MILLIS)
         HttpConnectionParams.setSoTimeout(params, TIMEOUT_MILLIS)
 
         int statusCode
-        def responseEntity
+        String responseEntity
         try {
             HttpResponse response = httpClient.execute(httpPost)
-            statusCode = response.getStatusLine().getStatusCode()
-            responseEntity = EntityUtils.toString(response.getEntity(), "utf-8")
-        } catch (Exception e) {
+            statusCode = response.statusLine.statusCode
+            HttpEntity entity = response.entity
+            responseEntity = EntityUtils.toString(entity, "utf-8")
+        } catch (IOException | ParseException e) {
```

Comment:
One of the inspections was to make the caught exception less generic - I've done this where the code documents clearly what exception types can be thrown.